src/goto-symex/partial_order_concurrency.cpp

```diff
@@ -56,16 +56,15 @@ void partial_order_concurrencyt::add_init_writes(
        e_it->is_shared_read() ||
        !e_it->guard.is_true())
     {
-      init_steps.push_back(symex_target_equationt::SSA_stept());
+      init_steps.push_back(symex_target_equationt::SSA_stept(
+        e_it->source, goto_trace_stept::typet::SHARED_WRITE));
```

Comment:
How about `emplace_back` as is used below.

Reply:
done